addons/addon-base-raas/packages/base-raas-rest-api/lib/controllers/study-controller.js

```diff
@@ -167,6 +171,7 @@ async function configure(context) {
       }
 
       const result = await studyPermissionService.update(requestContext, studyId, updateRequest);
+      await environmentMountService.manageWorkspacePermissions(studyId, updateRequest);
```

Comment:
I think renaming the method to applyWorkspacePermissions() might make more sense.
Regarding the design,  it might be worth having a class diagram that shows the relationship between  studyPermissionService, environmentMountService and environmentScService.

Reply:
Sounds good. I'll work on the class diagram and add it to the Quip doc.